inc/wp-components/classes/class-component.php

```diff
@@ -182,62 +241,82 @@ public function set_children( array $children, $append = false ) : self {
+			$this->component_groups[ $group ] = array_merge( $this->component_groups[ $group ], array_filter( $children ) );
+		} elseif ( $this->is_component( $group ) ) {
+			$this->children = array_merge( $this->children, array_filter( $group ) );
 		}
+
 		return $this;
 	}
 
 	/**
-	 * Prepend an array of components to the children array.
+	 * Prepend an array of components to the children array or component group.
 	 *
-	 * @param array $children Array of components.
+	 * @param string|array $group Group or array of components.
+	 * @param array        $children Child components.
 	 * @return self
 	 */
-	public function prepend_children( array $children ) : self {
-		if ( ! empty( $children ) ) {
-			$children = array_filter( $children );
-			$this->children = array_merge( $children, $this->children );
+	public function prepend_children( $group, array $children = [] ) : self {
+		// Prepend to group if group is valid, otherwise prepend to children.
+		if ( $this->is_valid_group( $group ) && ! empty( $children ) ) {
+			$this->component_groups[ $group ] = array_merge( array_filter( $children ), $this->component_groups[ $group ] );
+		} elseif ( $this->is_component( $group ) ) {
+			$this->children = array_merge( array_filter( $group ), $this->children );
 		}
+
 		return $this;
 	}
 
 	/**
-	 * Append a component to the children array.
+	 * Append a component to the children array or component group.
 	 *
-	 * @param Component $child Child component.
+	 * @param string|Component $group Group or array of components.
+	 * @param bool|Component   $child Child component.
 	 * @return self
 	 */
-	public function append_child( $child ) : self {
-		if ( ! empty( $child ) ) {
-			if ( is_array( $child ) ) {
-				$child = $child[0];
-			}
-			array_push( $this->children, $child );
+	public function append_child( $group, $child = false ) : self {
+		if ( is_array( $child ) ) {
+			$child = $child[0];
 		}
+
+		// Push to group if group is valid, otherwise push to children.
+		if ( $this->is_valid_group( $group ) && ! empty( $child ) ) {
+			array_push( $this->component_groups[ $group ], $child );
+		} elseif ( $this->is_component( $group ) ) {
+			array_push( $this->children, $group );
+		}
+
 		return $this;
 	}
 
 	/**
-	 * Prepend a component to the children array.
+	 * Prepend a component to the children array or component group.
 	 *
-	 * @param Component $child Child component.
+	 * @param string|Component $group Group or array of components.
+	 * @param bool|Component   $child Child component.
 	 * @return self
 	 */
-	public function prepend_child( $child ) : self {
-		if ( ! empty( $child ) ) {
-			if ( is_array( $child ) ) {
-				$child = $child[0];
-			}
-			array_unshift( $this->children, $child );
+	public function prepend_child( $group, $child = [] ) : self {
+		if ( is_array( $child ) ) {
+			$child = $child[0];
 		}
+
+		// Push to group if group is valid, otherwise push to children.
```

Comment:
Again, not a blocker. With PHP, push appends while unshift prepends. So comment here is misleading.